### PR TITLE
Remote-run: remove platform warning

### PR DESCRIPTION
**Pull request by JackuB**
After this PR, cifuzz now let's you bundle and remote-run Gradle and Maven projects without limitations

TODO: let's refactor the feature/support detection into the `config.go`, so we can control the experience and support matrix more clearly ▶️ https://code-intelligence.atlassian.net/browse/CLI-893
TODO: Working on tests

### CODE REVIEW

**Comment by jochil** [comment]
sorry for the missing task description... that was misleading. In the context of the jazzer bundling we do not want to remove these warnings completely, just the "WARNING: Bundling for %s is experimental and is currently not supported for remote runs." and only for jazzer.
We still do not support bundling on `windows` and `darwin` for build systems other than `maven` and `gradle`. And the `CIFUZZ_BUNDLE_ON_UNSUPPORTED_PLATFORMS` env is still needed for bundling cmake on mac for example.

So basically it is:
Java (Gradle|Maven) + Windows, Mac, Linux => no warning, bundle possible
C/CPP (bazel, cmake, other) + Linux => no warning, bundle possible
C/CPP (bazel, cmake, other) + Windows, Mac => warning, bundling only possible with `CIFUZZ_BUNDLE_ON_UNSUPPORTED_PLATFORMS`

Sorry for the confusion

**Comment by MarkusZoppelt** [comment]
Something seems wrong, I still get this on macOS:

```
> CIFUZZ_BUNDLE_ON_UNSUPPORTED_PLATFORMS=1 cifuzz remote-run -C examples/maven com.example.FuzzTestCase
Starting a remote run is currently only supported on Linux. If you are
interested in using this feature on macOS, please file an issue at
https://github.com/CodeIntelligenceTesting/cifuzz/issues
```

**Comment by MarkusZoppelt** [comment]
> https://app.code-intelligence.com/dashboard/projects/locall-e1af0159/campaign_runs/194cffea-5a3b-497d-b0cb-30309f4bfd54/overview%3Forigin=cli

Looks like the `origin=cli` URL parameter is still garbled in remote-run.
Fix here: #625

File: internal/cmd/remote-run/remote_run.go [hunk 1]
```diff
@@ -106,26 +106,27 @@ variable or by running 'cifuzz login' first.
...
+			// Fail early if the platform is not supported
+			if runtime.GOOS != "linux" {
+				if !sliceutil.Contains([]string{config.BuildSystemMaven, config.BuildSystemGradle}, opts.BuildSystem) {
+					system := cases.Title(language.Und).String(runtime.GOOS)
+					if runtime.GOOS == "darwin" {
+						system = "macOS"
+					}
+					err := errors.Errorf(`Starting a remote run is currently only supported on Linux. If you are
+				interested in using this feature on %s, please file an issue at
+				https://github.com/CodeIntelligenceTesting/cifuzz/issues`, system)
```
Comment (MarkusZoppelt) [inline internal/cmd/remote-run/remote_run.go:124]: Fix indentation

```suggestion
interested in using this feature on %s, please file an issue at
https://github.com/CodeIntelligenceTesting/cifuzz/issues`, system)
```
